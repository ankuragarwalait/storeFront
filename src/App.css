/* Global Element Styles ------------------------------------------------------------------------------*/
body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.App {
  text-align: center;
}

hr {
  margin: 1rem 0;
}

img.img-responsive {
  width: 100%;
  max-width: 100%;
}

.btn {
  display: block;
  padding: 1rem 3rem;
  font-size: 0.75rem; 
  color: white;
  border-width: 0;
  cursor: pointer;
  text-decoration: none;
}

.btn-primary {
  background-color: black;
}

.btn-secondary {
  background-color: grey;
}

.btn-tertiary {
  background-color: whote;
  color: black;
  border: 2px solid black;
}

.caret-link {
  cursor: pointer;
}

.caret-link::after {
  content: "";
  display: inline-block;
  margin-left: 0.2rem;
  vertical-align: middle;
  border-left: 0.2rem solid transparent;
  border-right: 0.2rem solid transparent;
  border-top: 0.2rem solid black;
}

/* Grid Styles ------------------------------------------------------------------------------ */

.container {
  max-width: 75rem;
  margin: 0 auto;
}

.row{
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  justify-content: space-between;
}

.col-1 {
  flex: 1 0 8.33%;
}

.col-2 {
  flex: 1 0 16.66%;
}

.col-4 {
  flex: 1 0 30%;
  margin-right: 3.33%;
}

.col-6 {
  flex: 1 0 50%;
}

.col-8 {
  flex: 1 0 66.66%;
}

.col-4:nth-of-type(3n+3) {
  margin-right: 0;
}

/* Alignment Styles ------------------------------------------------------------------------------*/

.justify-center {
  justify-content: center;
}

.align-center {
  align-items: center;
}

.justify-around {
  justify-content: space-around;
}

.text-left {
  text-align: left;
}

.text-right {
  text-align: right;
}

/* Typography and Theme ------------------------------------------------------------------------------*/

.t-price, .t-brand {
  color: gray;
}

.t-brand {
  font-size: .75rem;
  font-weight: 500;
}

.t-font-xs {
  font-size: .25rem;
}

.t-font-s {
  font-size: .5rem;
}

.t-font-m {
  font-size: .75rem;
}

.t-cancel {
  float: right;
  cursor: pointer;
  color: gray;
  font-weight: bold;
}

.t-border-grey {
  border-color: grey;
}

.t-border-light-grey {
  border-color: lightgrey;
}

.t-color-black {
  color: black;
}

.t-color-white {
  color: white;
}

.t-color-grey {
  color: grey;
}

.t-color-lightgrey {
  color: lightgrey;
}
