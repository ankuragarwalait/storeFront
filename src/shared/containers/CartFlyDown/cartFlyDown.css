.cart-flydown {
    padding: 1rem;
    position: absolute;
    width: 23rem;
    right: 1rem;
    background-color: white;
    z-index: 100;
    box-shadow: -.125rem .125rem .25rem lightgrey; 
}

.cart-prod-tile {
    margin-bottom: 1rem;
}

.cart-prod-tile:last-of-type {
    margin-bottom: 0;
}

.cart-prod-tile img {
    max-height: 6rem;
}

.cart-flydown p{
    margin-top: 0;
    margin-bottom: .35rem;
}