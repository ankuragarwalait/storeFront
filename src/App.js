import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Category from "./category/Category";
import Cart from "./cart/Cart";
import Product from "./product/Product";
import Header from './shared/containers/Header';

class App extends Component {
  render() {
    return (
      <div className="App">    
        <Header cartCount={0} />
        <Switch>
          <Route exact path="/" component={Category} />
          <Route path="/cart" component={Cart}/>
          <Route path="/product/:id" component={Product}/>
        </Switch>
      </div>
    );
  }
}

export default App;
