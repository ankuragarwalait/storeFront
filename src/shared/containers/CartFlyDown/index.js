import React from 'react'
import './cartFlyDown.css';
import CartFlyDownTile from '../../components/CartFlyDownTile';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removeFromCart } from '../../../store/actions/cartActions';
import PropTypes from 'prop-types';

const CartFlyDown = ({
  prodList,
  cartTotal,
  removeFromCart
}) => {
    return (
        <div className="cart-flydown">
            {Object.keys(prodList).map(prodItem => 
              <CartFlyDownTile {...prodList[prodItem]} key={prodItem} removeFromCart={removeFromCart}/>
            )}
            <hr className="t-border-light-grey"/>
            <div className="row">
                <p>Total</p>
                <p>${cartTotal}</p>
            </div>
            <div className="row">
                <Link to="/cart" className="btn btn-tertiary">VIEW CART</Link>
                <button className="btn btn-primary">CHECKOUT</button>
            </div>
        </div>
    )
}

const mapStateToProps = ({ cart }) => {
  return {
    prodList: cart.items,
    cartTotal: parseFloat(cart.total).toFixed(2)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    removeFromCart: (id) => dispatch(removeFromCart(id))
  }
}

CartFlyDown.propTypes = {
  prodList: PropTypes.objectOf(PropTypes.shape({
      id: PropTypes.string,
      title: PropTypes.string,
      image: PropTypes.string,
      brand: PropTypes.string,
      description: PropTypes.string,
      price: PropTypes.number
  })),
  cartTotal: PropTypes.string,
  removeFromCart: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(CartFlyDown);
