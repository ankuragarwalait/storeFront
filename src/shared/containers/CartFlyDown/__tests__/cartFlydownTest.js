import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CartFlyDown from '..';
import { mockStore } from '../../../../App.test';


let tree = null;
let store = null;

describe('test connected cart flydown component', () => {
    beforeEach(()=> {
        store = mockStore;
    });

    it('cart flydown component rendered snapshot', ()=> {
        tree = renderer.create(<BrowserRouter>
            <Provider store={store}>
                <CartFlyDown />
            </Provider>
            </BrowserRouter>);
        const prodJson = tree.toJSON();
        expect(prodJson).toMatchSnapshot();
    });  
});
