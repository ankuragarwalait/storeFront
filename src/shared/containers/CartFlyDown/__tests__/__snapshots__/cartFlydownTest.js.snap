// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test connected cart flydown component cart flydown component rendered snapshot 1`] = `
<div
  className="cart-flydown"
>
  <div
    className="row cart-prod-tile"
  >
    <div
      className="col-4 "
    >
      <img
        alt="Mashiko-Yaki Indigo Small Plate"
        className="img-responsive"
        src="/media/mashiko-yaki-indigo-small-plate.jpg"
      />
    </div>
    <div
      className="col-8 text-left t-font-m"
    >
      <span
        className="t-cancel"
        onClick={[Function]}
      >
        ✕
      </span>
      <p>
        Mashiko-Yaki Indigo Small Plate
      </p>
      <p
        className="t-font-m"
      >
        x
        2
      </p>
      <p
        className="t-brand"
      >
        Kiriko
      </p>
      <p>
        $
        20.00
      </p>
    </div>
  </div>
  <div
    className="row cart-prod-tile"
  >
    <div
      className="col-4 "
    >
      <img
        alt="Mashiko-Yaki Saucer"
        className="img-responsive"
        src="/media/mashiko-yaki-saucer.jpg"
      />
    </div>
    <div
      className="col-8 text-left t-font-m"
    >
      <span
        className="t-cancel"
        onClick={[Function]}
      >
        ✕
      </span>
      <p>
        Mashiko-Yaki Saucer
      </p>
      <p
        className="t-font-m"
      >
        x
        1
      </p>
      <p
        className="t-brand"
      >
        Kiriko
      </p>
      <p>
        $
        10.00
      </p>
    </div>
  </div>
  <hr
    className="t-border-light-grey"
  />
  <div
    className="row"
  >
    <p>
      Total
    </p>
    <p>
      $
      30.00
    </p>
  </div>
  <div
    className="row"
  >
    <a
      className="btn btn-tertiary"
      href="/cart"
      onClick={[Function]}
    >
      VIEW CART
    </a>
    <button
      className="btn btn-primary"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
